pkgs/by-name/yk/ykush/package.nix

```
@@ -0,0 +1,50 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  libusb1,
+  runCommand,
+  gcc,
+}:
+stdenv.mkDerivation (final: {
+  pname = "ykushcmd";
+  version = "1.3.0";
+
+  buildInputs = [ libusb1 ];
+  nativeBuildInputs = [ gcc ];
+
+  src = fetchFromGitHub {
+    owner = "yepkit";
+    repo = "ykush";
+    rev = "v${final.version}";
+    hash = "sha256-ykZAuHk3zPiH20yMYPgq6kJv1LAhUko6jJfAqGREvWA=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D bin/ykushcmd $out/bin/ykushcmd
+    install -Dm444 ${./99-ykush.rules} $out/lib/udev/rules.d/99_ykush.rules
+
+    runHook postInstall
+  '';
+
+  passthru.tests = {
+    # We can't run any actual tests without hardware, but we can at least check the binary.
+    run-only = runCommand "${final.pname}-test" ''
+      ${final}/bin/ykushcmd -h | grep YKUSHCMD
+    '';
+  };
+
+  meta = {
+    description = "Control utility for YepKit USB hubs.";
+    longDescription = ''
+      contols the features of YepKit USB hubs, allowing one to arbitrarily
+      turn ports on and off and otherwise control the hub.
+    '';
+    homepage = "https://github.com/Yepkit/ykush";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ ktemkin ];
+    mainProgram = "ykushcmd";
```

```suggestion
    mainProgram = "ykushcmd";
    platforms = lib.platforms.linux;
```

Remember `stdenv` doesn't set `platforms` automatically.

---

resolved in https://github.com/NixOS/nixpkgs/commit/6ba9011410f5f22c49ddd82304939e4d7af22c05 in https://github.com/NixOS/nixpkgs/pull/469647

https://github.com/NixOS/nixpkgs/pull/469647/files#diff-e089a30b2f6df2cb181eea2621aba8ec323735c8bf851166f0558653486e24e0R50